### PR TITLE
Adds in energy pistols

### PR DESCRIPTION
**Pull request by shazbot194**
h-kr-po1rsgzmd6x5fjgog
Hello Nanotrasen, so I know our last pitch wasn't quite a success, since it appears you do have a service pistol, however, that pistol is very rare and is custom made for order. We would like to take that design and start to mass produce it, making it available for more general purpose ordering. We do hope this is a much more tempting offer.

//Transmission over//

Well, since so many people showed support for the mini, I'm putting my sprites on ice and just using the current pistol sprite and code, with some slight alterations, namely, it now has slightly less ammo, 100 less energy to put it within the original lines of this PR, which is 5 lethal shots, and 10 disabler shots. I also added item states to show how much ammo the pistol has. The change to the mini also means that now these pistols now sport a flashlight, which has to be right clicked to enable. I'll most likely change this to the normal icon based system in another PR as it is slightly outside the scope of this PR, and is worth adding by itself. This still adds in the a crate to order them and replaces the HoP's E-gun with a mini. 

🆑 Shazbot
Adds: Adds E-pistol crates to the cargo manifest
Change: changes the HoP's E-gun to an E-pistol
/🆑

### CODE REVIEW

**Comment by StreakyHaddock** [comment]
Oh this is neat. Even if it has less capacity, I'd like one just for style points.

**Comment by shazbot194** [comment]
Its always nice to hear people enjoying my sprites. I would carry one for both style points, and because it would be easier, as I rarely use the lethal option anyway.

**Comment by Tayyyyyyy** [comment]
Really cool! Just a minor thing, should the inhand sprite show whether it's on lethal or disable? The normal egun does that and it can be helpful.

File: code/game/objects/structures/crates_lockers/closets/secure/security.dm [hunk 1]
```diff
@@ -52,7 +52,7 @@
 	new /obj/item/weapon/storage/box/ids(src)
 	new /obj/item/weapon/storage/box/PDAs(src)
 	new /obj/item/clothing/suit/armor/vest(src)
-	new /obj/item/weapon/gun/energy/gun(src)
+	new /obj/item/weapon/gun/energy/gun/pistol(src)
```
[the remark this answers is not in the document]
Comment (alexkar598) [reply code/game/objects/structures/crates_lockers/closets/secure/security.dm]: thats the hop(Head of Personnel) not hos

[the remark this answers is not in the document]
Comment (shazbot194) [reply code/game/objects/structures/crates_lockers/closets/secure/security.dm]: This is the HoP, not HoS's locker

[the remark this answers is not in the document]
Comment (Spartan6) [reply code/game/objects/structures/crates_lockers/closets/secure/security.dm]: 'secure/security/'
I'm a dumbass and misread.
Yeah this works. Captain's gun should also be changed none the less. AAAAAAAAAA

**Comment by Fox-McCloud** [comment]
While the idea of replacing the HoP's gun may be worth debating....

I'm not sure this gun is needed.

There's one that already exists, and the sprite is, in my opinion, better:

img

**Comment by Spartan6** [comment]
I'll agree the sprite that Fox supplied does appear to look more in line with our current weaponary.

**Comment by shazbot194** [comment]
I can change it back if people want, I did ask a couple people, and they seemed to like the new sprite more.

**Comment by Ty-Omaha** [comment]
>After some more testing, I made a new sprite which I think is more in line, for this one, use the confused emote.

That looks like an SMG / Carbine

**Comment by shazbot194** [comment]
personally reminds me of some kind of heavy pistol

**Comment by Cactus2388** [comment]
The HoP doesn't need a full egun to "fight off" intruders. We have roles like blueshield and they do well with less firepower. None the less this'll stop potential "vigilante" HOPs

**Comment by Necaladun** [comment]
Might be worthwhile for a HoP replacement for sure (although I'd prefer something non lethal entirerly), but otherwise I'm not sure what niche these guns would fill. Security don't need weapons that go in their pockets, and I'm not sure why you'd want one of these over a normal egun, except to put in your pocket. 

The sprites Fox showed there I am much more in favour of, but a cool sprite alone isn't enough to justify adding these.

**Comment by shazbot194** [comment]
Well the nitch they would fill would be the same that any pistol fills, it's a weaker, easier to carry, cheaper to make firearm. Here that wouldn't mean that it would be cheaper to arm people with this in cases like blob, nukies, ext, but any person trying to use two pistols vs 1 carbine would suffer in combat performance. Thus they would have a small nitch to fill, in addition if you didn't trust someone with a full E-gun, you couldn't give them this, such as the RD or CMO when they are believed to never assasination targets.

And due to the realitively large support for the old sprite, I'm going to put mine on ice for now, and just modify the mini to have the same cell size, and use if rather then my custom pistol. This will mean the pistol is slightly better because it would have a flashlight, but this is a small change at best.

**Comment by StreakyHaddock** [comment]
Quite honestly, as Blueshield I'd love to have little sidearms to give to vulnerable VIPS (Magistrate, NT Rep, CE, CMO) since that's what I do with disabler crates now.
As a Sec Officer, I'd love having a couple of backup sidearms that I can fit in a box.

**Comment by shazbot194** [comment]
Well, I made the shift yesterday, but I should probably make a comment so people can see that I have changed from the old sprites and custom E-pistol to just using the mini, as Fox had suggested. All I did was change the max cell size to 500, which is half of the E-guns 1000, and what I originally had for the E-pistol. In addition to adding some in hand sprites, so now it looks like your person is actually holding the gun, and not some generic pistol.

**Comment by Vivalas** [comment]
I think we have enough guns... and making a new gun just to replace what the HoP has is a bit silly.

**Comment by shazbot194** [comment]
Currently, this PR isn't adding in a new gun, but rather using the Mini, which is a gun already in game, just unused.

**Comment by Spartan6** [comment]
This isn’t adding a new gun, more than updating and implementing.

**Comment by Denthamos** [comment]
I like the idea of this.  A non lethal only version would be nice as well for when like streaky said certain VIP's or people being targeted need a form of defense which disablers are used for at the moment.

File: code/modules/projectiles/guns/energy/nuclear.dm [hunk 2]
```diff
@@ -39,8 +38,8 @@
 /obj/item/weapon/gun/energy/gun/mini/New()
 	gun_light = new /obj/item/device/flashlight/seclite(src)
 	..()
-	power_supply.maxcharge = 600
-	power_supply.charge = 600
+	power_supply.maxcharge = 500
+	power_supply.charge = 500
```
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/energy/nuclear.dm]: leave these values alone, please.

File: code/datums/supplypacks.dm [hunk 3]
```diff
@@ -388,6 +388,15 @@ var/list/all_supply_groups = list(supply_emergency,supply_security,supply_engine
 	containertype = /obj/structure/closet/crate/secure/plasma
 	containername = "energy gun crate"
 
+/datum/supply_packs/security/armory/epistol	// costs 3/5ths of the normal e-guns for 3/4ths the total ammo, making it cheaper to arm more people, but less convient for any one person
+	name = "Energy Pistol Crate"
+	contains = list(/obj/item/weapon/gun/energy/gun/mini,
+					/obj/item/weapon/gun/energy/gun/mini,
+					/obj/item/weapon/gun/energy/gun/mini)
+	cost = 15
```
Comment (Fox-McCloud) [inline code/datums/supplypacks.dm]: Good cost if the values, below, are left alone.

**Comment by chaoko99** [comment]
image

>>>Literally a prop from alien.

File: code/modules/projectiles/guns/energy/nuclear.dm [hunk 4]
```diff
@@ -30,7 +30,6 @@
 	name = "miniature energy gun"
 	desc = "A small, pistol-sized energy gun with a built-in flashlight. It has two settings: stun and kill."
 	icon_state = "mini"
-	item_state = "gun"
```
Comment (tigercat2000) [inline code/modules/projectiles/guns/energy/nuclear.dm]: Why did you change this?

**Comment by tigercat2000** [comment]
Fix the conflicts and I'll merge this.

File: code/modules/projectiles/guns/energy/nuclear.dm (hunk 4 above)
Comment (shazbot194) [reply code/modules/projectiles/guns/energy/nuclear.dm]: Because I made my own in hand sprites that work much better.

Comment (tigercat2000) [reply code/modules/projectiles/guns/energy/nuclear.dm]: This is dealt with by a parent*
